themes/default/content/docs/reference/yaml/_index.md

```diff
@@ -82,6 +82,54 @@ The optional `customTimeouts` property of a resource is an object of the followi
 | `delete`      | string | No | No | Delete is the custom timeout for delete operations. |
 | `update`      | string | No | No | Update is the custom timeout for update operations. |
 
+### Providers and provider versions
+
+There are at least two reasons to explicitly define providers in YAML, or explicitly set their versions while creating resources.
+
+* First, using explicit versions (aka provider pinning) avoids the problem that a deployment can suddenly fail because the provider's behavior has changed with a newer version.
+* Second, using default providers in Pulumi can be problematic, as described very well in the blog article [Unlock Programmatic Control by Disabling Default Providers]({{< relref "/blog/disable-default-providers/index" >}}).
```

Comment:
nit: I'd want to change this to more positive framing :) 

Perhaps we could say:

```suggestion
1. Using explicit versions enables pinning the dependencies used, a technique used to improve build reliability.
1. Using explicit providers enables controlling the options for providers used by each resource, as described in [Unlock Programmatic Control by Disabling Default Providers]({{< relref "/blog/disable-default-providers/index" >}}).
```

Reply:
@AaronFriel  I'm fine with the changes. As I'm a non native, it's always a bit tricky for me to find the correct wording.

Reply:
@tmeckel no worries at all!! we very much appreciate you taking the time to open a pr ❤️